### PR TITLE
PostgreSQL 10 support

### PR DESCRIPTION
**Pull request by ioguix**

### CODE REVIEW

File: script/pgsqlms [hunk 1]
```diff
@@ -74,6 +74,186 @@ my $ATTRD_PRIV    = "$HA_SBIN_DIR/attrd_updater --private --lifetime reboot";
...
+# numeric pgsql versions
+my $PGVERNUM;
+my $PGVER_10  = 100000;
+
+# Run a query using psql.
+#
+# This function returns an array with psql return code as first element and
+# the result as second one.
+#
+sub _query {
```
Comment (ioguix) [inline script/pgsqlms]: I just moved the sub "_query" here, no code modification, you can jump this sub in your review.

File: script/pgsqlms [hunk 2]
```diff
@@ -74,6 +74,186 @@ my $ATTRD_PRIV    = "$HA_SBIN_DIR/attrd_updater --private --lifetime reboot";
...
+    #   2: unable to connect
+    #   3: query failed
+    return $rc;
+}
+
+# Get the last received location on a standby
+# if the first argument is true, returns the value as decimal
+# if the first argument is false, returns the value as LSN
+# Returns undef if query failed
+sub _get_last_received_lsn {
```
Comment (ioguix) [inline script/pgsqlms]: I created this sub tho "hide" the mess with pg10 compat around the query from the callers.

File: script/pgsqlms [hunk 3]
```diff
@@ -74,6 +74,186 @@ my $ATTRD_PRIV    = "$HA_SBIN_DIR/attrd_updater --private --lifetime reboot";
...
+    ocf_log( 'err', 'No values for last received LSN' )
+        if $rc == 0 and not $rs[0][0];
+
+    return undef;
+}
+
+# Get the master score for each connected standby
+# Returns directly the result set of the query or exit with an error.
+# Exits with OCF_ERR_GENERIC if the query failed
+sub _get_lag_scores {
```
Comment (ioguix) [inline script/pgsqlms]: Same here, I created this sub tho "hide" the mess with pg10 compat around the query from the callers.

File: script/pgsqlms [hunk 4]
```diff
@@ -1060,7 +1105,6 @@ sub pgsql_validate_all {
     my $fh;
     my $ans = '';
     my $PGVERSION;
-    my $PGVERNUM;
```
Comment (ioguix) [inline script/pgsqlms]: moved as a global var

**Comment by ioguix** [comment]
I reviewed my own code twice so far, but I would appreciate some more eyes on this patch.

For your information, I did the following tests:

* 3 nodes cluster with pgsql 10
  - switchover
  - failover
  - failover with lagging designated master (election failed)
* 3 nodes cluster with pgsql 9.4
  - switchover
  - failover
  - failover with lagging designated master (election failed)

File: script/pgsqlms [hunk 5]
```diff
@@ -1173,15 +1217,21 @@ sub pgsql_validate_all {
...
     if ( $PGVERNUM < 90300 ) {
         ocf_exit_reason( "PostgreSQL %s not supported. Require 9.3 and more",
             $PGVERSION );
         exit $OCF_ERR_INSTALLED;
     }
 
+    $PGXLOGDUMP = "$bindir/pg_waldump" if $PGVERNUM >= $PGVER_10;
+
     # check binaries
     unless ( -x $PGCTL and -x $PGPSQL and -x $PGCTRLDATA and -x $PGISREADY ) {
```
Comment (blogh) [inline script/pgsqlms]: It's not related to this PR in particular but is there a reason not to check for $PGXLOGDUMP here ?

Comment (ioguix) [inline script/pgsqlms]: You are right, I'll add a commit to this branch related to this.

**Comment by ioguix** [comment]
About the message referencing xlogdump, I did not want to add a conditional block just for a debug message intended to devs.

However, I will edit the debug message to use the new name, no need to keep the old one, as far as devs understand waldump == xlogdump depending on the pgsql version.

Thank you for the quick review blogh!

**Pull request merged**